src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+        Objects.requireNonNull(anOptionName, "Given option is null.");
+        //throws IllegalArgumentException if the given name does not match a constant name in the enum
+        ChainFragmentLengthOption tmpConstant = ChainFragmentLengthOption.valueOf(anOptionName);
+        this.setChainFragmentLengthSetting(tmpConstant);
+    }
+    /**
+     * Sets the option for maximum length of fragment chain.
+     *
+     * @param anOption the option to use
+     * @throws NullPointerException if given option is null
+     */
+    public void setChainFragmentLengthSetting(ChainFragmentLengthOption anOption) throws NullPointerException {
+        Objects.requireNonNull(anOption, "Given option is null.");
+        this.chainFragmentLengthSetting.set(anOption.name());
+    }
+    /**
+     * Sets the bool option for spiro carbon dissection.
+     *
+     * @param aBoolean the bool option to use
+     */
+    public void setDissectSpiroCarbonSetting(boolean aBoolean) {
+        this.dissectSpiroCarbonSetting.set(aBoolean);
+    }
+    /**
+     * Sets the bool option for ring dissection.
+     *
+     * @param aBoolean the bool option to use
+     */
+    public void setDissectRingsSetting(boolean aBoolean) {
+        this.dissectRingsSetting.set(aBoolean);
+    }
+    /**
+     * Sets the enum option for maximum size of preserved ring.
+     *
+     * @param anOptionName the enum option to use
+     * @throws NullPointerException if given option is null
+     * @throws IllegalArgumentException if given option does not exist
+     */
+    public void setPreserveRingMaxSizeSetting(String anOptionName) throws NullPointerException, IllegalArgumentException {
+        Objects.requireNonNull(anOptionName, "Given option is null.");
+        PreserveRingMaxSizeOption tmpConstant = PreserveRingMaxSizeOption.valueOf(anOptionName);
+        this.setPreserveRingMaxSizeSetting(tmpConstant);
+    }
+    /**
+     * Sets the option for maximum size of preserved ring.
+     *
+     * @param anOption the option to use
+     * @throws NullPointerException if given option is null
+     */
+    public void setPreserveRingMaxSizeSetting(PreserveRingMaxSizeOption anOption) throws NullPointerException {
+        Objects.requireNonNull(anOption, "Given option name is null.");
+        this.preserveRingMaxSizeSetting.set(anOption.name());
+    }
+    /**
+     * Sets bool option for ring system preservation.
+     *
+     * @param aBoolean the bool option to use
+     */
+    public void setPreserveRingSystemSetting(boolean aBoolean) {
+        this.preserveRingSystemSetting.set(aBoolean);
+    }
+    /**
+     * Sets the maximum of rings contained in preserved system.
+     *
+     * @param anInt the int option to use
+     */
+    public void setPreserveRingSystemMaxSetting(int anInt) {
+        this.preserveRingSystemMaxSetting.set(anInt);
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="MORTAR Utilities">
```

Comment:
Non-sensical code fold description. We would usually call everything that is not a getter or setter "public methods".